src/business_logic/transaction/fee.rs

```diff
@@ -0,0 +1,133 @@
+use super::error::TransactionError;
+use crate::{
+    business_logic::{
+        execution::{
+            execution_entry_point::ExecutionEntryPoint,
+            gas_usage,
+            objects::{CallInfo, TransactionExecutionContext},
+        },
+        fact_state::{
+            in_memory_state_reader::InMemoryStateReader, state::ExecutionResourcesManager,
+        },
+        state::cached_state::CachedState,
+    },
+    definitions::general_config::{self, StarknetGeneralConfig},
+    services::api::contract_class::EntryPointType,
+};
+use felt::Felt;
+use num_traits::{Num, ToPrimitive};
+use std::cmp;
+use std::collections::{HashMap, HashSet};
+
+// ----------------------------------------------------------------------------------------
+/// Transfers the amount actual_fee from the caller account to the sequencer.
+/// Returns the resulting CallInfo of the transfer call.
+// ----------------------------------------------------------------------------------------
+
+pub(crate) fn execute_fee_transfer(
+    state: &mut CachedState<InMemoryStateReader>,
+    general_config: StarknetGeneralConfig,
+    tx_context: TransactionExecutionContext,
+    actual_fee: u64,
+) -> Result<CallInfo, TransactionError> {
+    if actual_fee as u64 > tx_context.max_fee {
+        return Err(TransactionError::FeeError(
+            "Actual fee exceeded max fee.".to_string(),
+        ));
+    }
+
+    let fee_token_address = general_config.starknet_os_config.fee_token_address.clone();
+
+    let calldata = [
+        general_config.block_info.sequencer_address.0.clone(),
+        Felt::from_bytes_be(&actual_fee.to_be_bytes()),
```

Comment:
Wasn't there an `impl From<u64> for Felt {}` available?

Reply:
i checked it out and yes, there was. i already changed it